FirebaseAppCheck/Sources/Public/FirebaseAppCheck/FIRAppCheckAvailability.h

```diff
@@ -16,17 +16,34 @@
 
 // Availability conditions for different App Check SDK components.
 
+#import <Foundation/Foundation.h>
 #import <TargetConditionals.h>
 
 #pragma mark - DeviceCheck
 
+// DeviceCheck availability was extended to watchOS in Xcode 14.
+#if defined(__WATCHOS_9_0) && __WATCH_OS_VERSION_MAX_ALLOWED >= __WATCHOS_9_0
+
+// Targets where DeviceCheck framework is available to be used in preprocessor conditions.
+#define FIR_DEVICE_CHECK_SUPPORTED_TARGETS \
+  TARGET_OS_IOS || TARGET_OS_OSX || TARGET_OS_TV || TARGET_OS_WATCH
+
+// `DeviceCheckProvider` availability.
+#define FIR_DEVICE_CHECK_PROVIDER_AVAILABILITY \
+  API_AVAILABLE(ios(11.0), macos(10.15), tvos(11.0), watchos(9.0))
```

Comment:
non-blocking: This is outside the scope of the review but I wonder if we should revisit the choice to specify `macos(10.15)` as the minimum version. It is the version specified by Apple but essentially it's a no-op on Catalina (`DCDevice.current.isSupported` returns false). I worry that it could mislead users into thinking they can use App Attest (using DeviceCheck) on macOS 10.15.

Reply:
Hmm. Interesting, I did not know that. Thanks for bringing that up. We could create an issue to track that for Firebase 10

Reply:
No problem! I've added an [FR](https://github.com/firebase/firebase-ios-sdk/issues/10095#issue-1336561148) with the details I know.

Note: IIRC, I did test DeviceCheck on iOS 11 but I haven't personally tried it on tvOS 11 (and I can't downgrade to test it). I would assume the non-macOS availability labels are accurate though.